api/v1_users_verify_token_test.go

```diff
@@ -0,0 +1,304 @@
+	assert.Equal(t, 400, status)
+}
+
+func TestUsersVerifyTokenMissingToken(t *testing.T) {
+	app := emptyTestApp(t)
+
+	// Test without token parameter
+	status, _ := testGet(t, app, "/v1/users/verify_token")
+	assert.Equal(t, 400, status)
+}
+
+func TestUsersVerifyTokenInvalidSignature(t *testing.T) {
+	app := emptyTestApp(t)
+
+	// Create a valid header and payload but invalid signature
+	header := map[string]interface{}{
+		"typ": "JWT",
+		"alg": "ES256K",
+	}
+	headerBytes, _ := json.Marshal(header)
+	base64Header := base64.URLEncoding.EncodeToString(headerBytes)
+	base64Header = strings.TrimRight(base64Header, "=")
+
+	payload := map[string]interface{}{
+		"userId": trashid.MustEncodeHashID(1),
+		"iat":    time.Now().Unix(),
+	}
+	payloadBytes, _ := json.Marshal(payload)
+	base64Payload := base64.URLEncoding.EncodeToString(payloadBytes)
+	base64Payload = strings.TrimRight(base64Payload, "=")
+
+	// Use invalid signature
+	invalidSignature := "invalidsignature"
+	token := fmt.Sprintf("%s.%s.%s", base64Header, base64Payload, invalidSignature)
+
+	status, _ := testGet(t, app, "/v1/users/verify_token?token="+token)
+	assert.Equal(t, 400, status)
+}
+
+func TestUsersVerifyTokenWalletNotFound(t *testing.T) {
+	app := emptyTestApp(t)
+
+	// Create test private key but don't add user to database
+	privateKey, err := crypto.GenerateKey()
+	assert.NoError(t, err)
+
+	payload := map[string]interface{}{
+		"userId": trashid.MustEncodeHashID(1),
+		"iat":    time.Now().Unix(),
+	}
+
+	token, err := createTestJWT(privateKey, payload)
+	assert.NoError(t, err)
+
+	status, _ := testGet(t, app, "/v1/users/verify_token?token="+token)
+	assert.Equal(t, 404, status)
+}
+
+func TestUsersVerifyTokenUnauthorizedWallet(t *testing.T) {
+	app := emptyTestApp(t)
+
+	// Create two different private keys
+	user1PrivateKey, err := crypto.GenerateKey()
+	assert.NoError(t, err)
+	user1Wallet := crypto.PubkeyToAddress(user1PrivateKey.PublicKey)
+	user1WalletLower := strings.ToLower(user1Wallet.Hex())
+
+	user2PrivateKey, err := crypto.GenerateKey()
+	assert.NoError(t, err)
+	user2Wallet := crypto.PubkeyToAddress(user2PrivateKey.PublicKey)
+	user2WalletLower := strings.ToLower(user2Wallet.Hex())
+
+	fixtures := database.FixtureMap{
+		"users": {
+			{
+				"user_id": 1,
+				"wallet":  user1WalletLower,
+				"handle":  "user1",
+			},
+			{
+				"user_id": 2,
+				"wallet":  user2WalletLower,
+				"handle":  "user2",
+			},
+		},
+	}
+
+	database.Seed(app.pool, fixtures)
+
+	// Create JWT payload for user 1 but sign with user 2's key (without manager relationship)
+	payload := map[string]interface{}{
+		"userId": trashid.MustEncodeHashID(1),
+		"iat":    time.Now().Unix(),
+	}
+
+	token, err := createTestJWT(user2PrivateKey, payload)
+	assert.NoError(t, err)
+
+	status, _ := testGet(t, app, "/v1/users/verify_token?token="+token)
+	assert.Equal(t, 403, status)
+}
+
+func TestUsersVerifyTokenInvalidPayload(t *testing.T) {
```

Comment:
Really appreciate the thorough tests. Gold star for you.

Reply:
gold star for anthropic